src/Engine/Graphics/IRender.h

```diff
@@ -341,7 +342,14 @@ class IRender {
     virtual void SavePCXScreenshot() = 0;
     virtual unsigned short* MakeScreenshot16(int width, int height) = 0;
 
-    virtual int GetActorsInViewport(int pDepth) = 0;
+    /**
+     * Find all visible actors (excluding dead ones or summoned by player)
+     * and call callback function for each actor found.
+     *
+     * @param pDepth    Depth of actors search.
+     * @param callback  Callback function.
+     */
+    virtual int forEachActorInViewport(int pDepth, std::function<void(Actor&)> callback) = 0;
```

Comment:
Ouch. This wasn't what I was thinking about tbh. If we're doing this for efficiency, then I have my doubts that a solution with a type-erased callback will be more efficient than just returning a vector of ids.

Reply:
Let's discuss some more in Discord.